### PR TITLE
Added support for pylas

### PR DESCRIPTION
**Pull request by SBCV**
This PR adds support for `pylas` and handle cases, when color information is missing in the las/laz file.
Changed the default backend to `pylas`, since it is much faster than `laspy`.

### CODE REVIEW

File: pyntcloud/io/las.py [hunk 1]
```diff
@@ -11,26 +15,64 @@ def convert_location_to_dtype(data, dtype_str):
...
 
 def get_color_dtype(data, column_names):
-    color_data_types = [data["points"][column_name].dtype for column_name in column_names]
-    assert len(set(color_data_types)) == 1
-    return color_data_types[0]
+    has_color = True
+    for column_name in column_names:
+        if column_name not in data["points"]:
+            has_color = False
+            break
```
Comment (daavoo) [inline pyntcloud/io/las.py:22]:
```suggestion
    has_color = all(column in data["points"] for column in column_names)
```

File: pyntcloud/io/las.py [hunk 2]
```diff
@@ -11,26 +15,64 @@ def convert_location_to_dtype(data, dtype_str):
...
+    if has_color:
+        color_data_types = [data["points"][column_name].dtype for column_name in column_names]
+        assert len(set(color_data_types)) == 1
```
Comment (daavoo) [inline pyntcloud/io/las.py:25]: Try to replace assertions with a concrete exception raising in order to inform the user what exactly went wrong.

Ref on reasons behind:
https://stackoverflow.com/a/20076295/6076911

File: pyntcloud/io/las.py [hunk 3]
```diff
@@ -11,26 +15,64 @@ def convert_location_to_dtype(data, dtype_str):
...
-    # Color information in las/laz files is stored as uint8 or uint16
-    assert input_dtype in ["uint8", "uint16"]
-    if input_dtype == "uint8" and output_dtype == "uint16":
-        data["points"].loc[:, column_names] *= 256
-    elif input_dtype == "uint16" and output_dtype == "uint8":
-        data["points"].loc[:, column_names] /= 256
-    data["points"] = data["points"].astype({"red": output_dtype, "green": output_dtype, "blue": output_dtype})
+    if input_dtype is not None:
+        # Color information in las/laz files is stored as uint8 or uint16
+        assert input_dtype in ["uint8", "uint16"]
```
Comment (daavoo) [inline pyntcloud/io/las.py:38]:
```suggestion
        if input_dtype not in ["uint8", "uint16"]:
            raise ValueError(f"Invalid color dtype. Expected one of ['uint8', 'uint16'] but got {input_dtype}")
```

File: pyntcloud/io/las.py [hunk 4]
```diff
@@ -46,17 +88,13 @@ def read_las(filename, xyz_dtype="float32", rgb_dtype="uint8"):
...
-        data = convert_location_to_dtype(data, xyz_dtype)
-        data = convert_color_to_dtype(data, rgb_dtype)
-        data["las_header"] = las.header
 
+    if backend == "laspy":
+        data = read_las_with_laspy(filename)
+    elif backend == "pylas":
+        data = read_las_with_pylas(filename)
+    else:
+        assert False
```
Comment (daavoo) [inline pyntcloud/io/las.py:97]: Replace with ValueError

**Review by daavoo — CHANGES_REQUESTED** [review]
Thanks for the P.R. @SBCV !! 

Overall looks very good, just minor comments to be addressed.

In addition to the comments, you would need to replace `laspy` in a few extra places in order to make `pylas` the default:

https://github.com/daavoo/pyntcloud/blob/master/requirements.txt#L4
https://github.com/daavoo/pyntcloud/blob/master/setup.py#L27
https://github.com/daavoo/pyntcloud/blob/master/binder/environment.yml#L17

File: pyntcloud/io/las.py (hunk 2 above)
Comment (SBCV) [reply pyntcloud/io/las.py:21]: Thank you for pointing that out!

**Comment by SBCV** [comment]
Addressed the requested changes and squashed the commit.

**Review by daavoo — APPROVED** [review]

**Comment by daavoo** [comment]
> Addressed the requested changes and squashed the commit.

Thank you very much!

**Comment by SBCV** [comment]
Quick question: Do you have any idea, when these changes will be available via pypi?
I would like to integrate these changes into [this Blender addon](https://github.com/SBCV/Blender-Addon-Photogrammetry-Importer) using `pip install pyntcloud`

**Comment by daavoo** [comment]
> Quick question: Do you have any idea, when these changes will be available via pypi?
> I would like to integrate these changes into [this Blender addon](https://github.com/SBCV/Blender-Addon-Photogrammetry-Importer) using `pip install pyntcloud`

Have to fix one thing and will release a new version before.this weel ends

**Comment by daavoo** [comment]
> > Quick question: Do you have any idea, when these changes will be available via pypi?
> > I would like to integrate these changes into [this Blender addon](https://github.com/SBCV/Blender-Addon-Photogrammetry-Importer) using `pip install pyntcloud`
> 
> Have to fix one thing and will release a new version before.this week ends

@SBCV v0.1.3 is on pypi

**Pull request merged**